Feature: Add the Not Found Page

HowardMak opened the pull request:
Add the Not Found Page:
- if users typed wrong address, the router will jump to the not found page
- the user has not logged in will be required to go the login page
- otherwise there will be a button for the user to go back to the previous page

Resolve: CP-15

Feat:
1. When users type address that does not exist whey will be redirected to the not found page
![image](https://github.com/user-attachments/assets/70ef6a86-7082-4680-b574-30a4f6df67e4)

2. If users have not logged in they will be require to go back to login page as they are only provided with "homepage" button

3. But if users have logged in, there will another button called "back" that users can go back to the previous page by clicking it.
![image](https://github.com/user-attachments/assets/51ff0e0e-5a1d-4a51-b9b8-864dc73c4896)
 
Fix:
public/images/404.jpg
public/images/bg-404.jpg
src/app/not-found.tsx

Commit:
Revert "Merge pull request #4 from alert-city/feat/CP-14-Registration-Page"

This reverts commit c1c3a57468bd1ecce5426e0e93500cb993d63b8f, reversing
changes made to 1102dfcf71c5ee0a604f28e9cf932092d11be0af.

HowardMak committed:
Feature: Add the Not Found Page

Add the Not Found Page:
- if users typed wrong address, the router will jump to the not found page
- the user has not logged in will be required to go the login page
- otherwise there will be a button for the user to go back to the previous page

Resolve: CP-15

danieltate888 commented on src/app/not-found.tsx:
```diff
@@ -0,0 +1,87 @@
+                    >
+                        Back
+                    </Button>
+                ) : (null)}
```
Can change to use logical AND operator
Like：loggedIn && <Button></Button>

danieltate888 commented on src/app/not-found.tsx:
```diff
@@ -0,0 +1,87 @@
+                    color="primary"
+                    onClick={() => router.push('/login')}
+                >
+                    HomePage
```
1. If the status is logged in, this HomePage button should not show
2. HomePage can change to Return to Login

danieltate888 commented on src/app/not-found.tsx:
```diff
@@ -0,0 +1,87 @@
+                <Button
+                    variant="contained"
+                    color="primary"
+                    onClick={() => router.push('/login')}
```
Please change the path "/login" from string to a variable. You should import the Path variable RouteConfig from "@/routes/route.ts" first.

danieltate888 commented on src/app/not-found.tsx:
```diff
@@ -0,0 +1,87 @@
+    const [loggedIn, setLoggedIn] = useState(false);
+
+    useEffect(() => {
+        const token = typeof window !== 'undefined' ? localStorage.getItem(AUTH_TOKEN) : null;
```
1. I have refactor the frontend's some variables, please use the new variables after I Merging to dev. AUTH_TOKEN to ACCESS_TOKEN.
2. I changed the logic of auth status. In previous, you trigger a router, if the refresh token is invalid, the response interceptor in the Apollo client will save the status into localStorage, and then after you triggering another a router, it will push to login page. But now, it will redirect directly in the Apollo client, so the AUTH_STATUS is redundant now.

danieltate888 commented on src/app/not-found.tsx:
```diff
@@ -0,0 +1,87 @@
+import { Button, Container, Typography, Box } from '@mui/material';
+import { useRouter } from 'next/navigation';
+import Image from 'next/image';
+import { AUTH_STATUS, AUTH_TOKEN } from '../shared/constants/storage';
```
You can use the path alias like: 
"@/shared/constants/storage"

danieltate888 reviewed (changes requested):
Please do it after mu new changes merged into dev.

danieltate888 committed:
Feat: Registration Page (#7)

Finalised following features:
- Moved favicon.ico from /app to public
- Added alertcity.png to public/images
- New Added middileware.ts to handle request and redirection
- Optimized the IndexConfig and RouteConfig
- Added Account Registration Page
- Added TwoFA.tsx
- Added ActivatePage.tsx
- Optimized the LoginForm.tsx

Resolve CP-14

Commit:
Feat: registration Page

Finalised following features:
- Moved favicon.ico from /app to public
- Added alertcity.png to public/images
- New Added middileware.ts to handle request and redirection
- Optimized the IndexConfig and RouteConfig
- Added Account Registration Page
- Added TwoFA.tsx
- Added ActivatePage.tsx
- Optimized the LoginForm.tsx

Resolve CP-14

Commit:
Feat: Welcome Information

Finalised following features:
- Added WelcomeSnackbar.tsx

Resolve CP-16

HowardMak committed:
feat: Add not found page

Add a not found page:
- Users put wrong addresses will be redirected to a not found page.
- Those who have not logged in will see a button called "return to login"
- Those who have logged in will see the button call "back"

Resolve: CP-15
